lib/azure/armrest/configuration.rb

```diff
@@ -0,0 +1,254 @@
+module Azure
+  module Armrest
+    class Configuration
+      # Used to store unique token information.
+      @tokens = Hash.new { |hash, key| hash[key] = [] }
+
+      # A lookup table for getting api-version strings per provider and service.
+      @providers = Hash.new { |hash, key| hash[key] = {} }
+
+      # Singleton methods for the class instance variables
+      class << self
+        attr_reader :tokens
+        attr_reader :providers
```

Comment:
rename `providers` (and `@providers`) to something distinguishable from the instance method.
Both `tokens` and `providers` should be private. You can use `Configuration.send` to access them from the service classes.